src/vbk/rpc_register.cpp

```diff
@@ -467,29 +467,46 @@ UniValue getrawpopmempool(const JSONRPCRequest& request)
 namespace {
 
 template <typename T>
-bool GetPayload(const typename T::id_t& hash, T& out, const Consensus::Params& consensusParams, uint256& hashBlock, const CBlockIndex* const block_index)
+bool GetPayload(const typename T::id_t& pid, T& out, const Consensus::Params& consensusParams, CBlockIndex** block_index)
 {
     LOCK(cs_main);
     auto& pop = VeriBlock::getService<VeriBlock::PopService>();
 
-    if (!block_index) {
-        auto& mp = pop.getMemPool();
-        auto* pl = mp.get<T>(hash);
-        if (pl) {
-            out = *pl;
-            return true;
+    auto& mp = pop.getMemPool();
+    auto* pl = mp.get<T>(pid);
+    if (pl) {
+        out = *pl;
+        return true;
+    }
+
+    // search in the alttree storage
+    const auto& alttree = pop.getAltTree();
+    const auto& payloads = alttree.getStorage().getContainingAltBlocks(pid.asVector());
+
+    for (const auto& p : payloads) {
+        auto* index = alttree.getBlockIndex(p);
+        if (index == nullptr) continue;
+
+        const auto& pids = index->template getPayloadIds<T>();
+        auto it = std::find(pids.begin(), pids.end(), pid);
+        if (it == pids.end()) continue;
+
+        auto* blockindexTmp = LookupBlockIndex(uint256(index->getHash()));
+        if (!blockindexTmp) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block hash not found");
         }
```

Comment:
also should be an assert.
If index says there's alt block with hash X, then both vBTC and alt-cpp must contain block X